src/utils/http.rs

```diff
@@ -55,15 +55,23 @@ impl<'b, const N: usize> Headers<'b, N> {
             .map(|(_, value)| value)
     }
 
-    pub fn set(&mut self, name: &'b str, value: &'b str) -> &mut Self {
+    pub fn try_set(&mut self, name: &'b str, value: &'b str) -> bool {
```

Comment:
Perhaps this should be a Result instead of a bool? I think `Result<&mut Self, ()>` would allow the headers to be added in a builder-style way, just how it works with the current `set`.

Reply:
Normally, when using a builder, you know how many headers you want to add and when it panics you can just fix the capacity. On the other hand, building up a set of headers e.g. by iterating through whatever httparse spits out doesn't really utilize a returned self reference, as it's hard to do in a loop :)

I can change it, because why not. Note to self: add an error type.